gateway/errors/errors.go

```diff
@@ -745,4 +746,9 @@ var Codes = errorCodeMap{
 		Description:    "This operation is not supported in LakeFS",
 		HTTPStatusCode: http.StatusMethodNotAllowed,
 	},
+	ERRLakeFSWrongEndpoint: {
+		Code:           "ERRLakeFSWrongEndpoint",
+		Description:    "S3 request received in UI handler, did you forget to set your s3 gateway domain name?",
```

Comment:
We can get the currently configured domain name from config, assuming we have access to it, to make this error a little friendlier (i.e. `S3 request received by UI handler, Did you configure your S3 client to use the lakeFS endpoint ("${s3_endpoint_url")?`)

Reply:
Done.